### PR TITLE
✨ feat/numeraire to base currency

### PR DESCRIPTION
**Pull request by simluyt**

### CODE REVIEW

File: src/AssetRegistry/MainRegistry.sol [hunk 1]
```diff
@@ -57,28 +57,28 @@ contract MainRegistry is Ownable {
...
-            ++numeraireCounter;
+            ++baseCurrencyCounter;
         }
     }
 
     /**
      * @notice Sets the new Factory address
      * @dev The factory can only be set on the Main Registry AFTER the Main registry is set in the Factory.
-     *      This ensures that the allowed Numeraires and corresponding stable contracts in both contract are equal.
+     *      This ensures that the allowed BaseCurrencys and corresponding stable contracts in both contract are equal.
```
Comment (Thomas-Smets) [inline src/AssetRegistry/MainRegistry.sol:81]: Currencys? :D

File: src/Liquidator.sol [hunk 2]
```diff
@@ -392,35 +392,35 @@ contract Liquidator is Ownable {
...
         uint256[] memory totalClaimable,
         address claimer
     ) internal {
-        for (uint8 k; k < numeraireCounter; ) {
+        for (uint8 k; k < baseCurrencyCounter; ) {
             if (totalClaimable[k] > 0) {
-                address numeraireStable = IFactory(factoryAddress)
-                    .numeraireToStable(k);
-                uint256 balance = IERC20(numeraireStable).balanceOf(
+                address baseCurrenciestable = IFactory(factoryAddress)
```
Comment (Thomas-Smets) [inline src/Liquidator.sol:405]: Think you replaced to much with last commit

File: src/AssetRegistry/FloorERC1155SubRegistry.sol [hunk 3]
```diff
@@ -39,9 +39,9 @@ contract FloorERC1155SubRegistry is SubRegistry {
      *                         - id: The Id of the asset
      *                         - assetAddress: The contract address of the asset
      *                         - oracleAddresses: An array of addresses of oracle contracts, to price the asset in USD
-     * @param assetCreditRatings The List of Credit Ratings for the asset for the different Numeraires.
-     * @dev The list of Credit Ratings should or be as long as the number of numeraires added to the Main Registry,
-     *  or the list must have length 0. If the list has length zero, the credit ratings of the asset for all numeraires is
+     * @param assetCreditRatings The List of Credit Ratings for the asset for the different BaseCurrencys.
```
Comment (Thomas-Smets) [inline src/AssetRegistry/FloorERC1155SubRegistry.sol:42]: Derde keer goede keer

File: src/Vault.sol [hunk 4]
```diff
@@ -86,11 +91,30 @@ contract Vault {
     modifier onlyFactory() {
         require(
             msg.sender == IMainRegistry(_registryAddress).factoryAddress(),
-            "VL: Not factory"
+            "VL: You are not the factory"
```
Comment (Thomas-Smets) [inline src/Vault.sol:94]: Add positive and negative test

File: src/Vault.sol [hunk 5]
```diff
@@ -86,11 +91,30 @@ contract Vault {
...
+     */
+    modifier onlyAuthorized() {
+        require(
+            allowed[msg.sender],
+            "VL: You are not authorized"
         );
         _;
     }
 
+    function authorize(address user) external onlyOwner {
```
Comment (Thomas-Smets) [inline src/Vault.sol:110]: Add tests

File: src/Vault.sol [hunk 6]
```diff
@@ -658,10 +680,30 @@ contract Vault {
...
-            numeraire
+            baseCurrency
         );
     }
 
+    /** 
+    @notice Sets the baseCurrency of a vault.
+    @dev First checks if there is no locked value. If there is no value locked then the baseCurrency gets changed to the param
+  */
+    function setBaseCurrency(uint8 newBaseCurrency) public onlyAuthorized onlyOwner {
```
Comment (Thomas-Smets) [inline src/Vault.sol:691]: Add tests for positive and negative scenarios

File: src/test/8_Vault.t.sol [hunk 7]
```diff
@@ -1624,19 +1622,81 @@ contract vaultTests is Test {
...
+
+        vm.startPrank(nonAuthorized);
+        vm.expectRevert("VL: You are not authorized");
+        vault_m.setBaseCurrency(Constants.EthBaseCurrency);
+        vm.stopPrank();
+
+        (,,,,,uint256 _baseCurrency) = vault_m.debt();
+        assertEq(_baseCurrency, Constants.UsdBaseCurrency);
+    }
+
```
Comment (Thomas-Smets) [inline src/test/8_Vault.t.sol:1699]: Add test when vault already has debt (think best to use a cheatcode here, maybe see it with Jasper)

File: src/Vault.sol [hunk 8]
```diff
@@ -86,11 +91,30 @@ contract Vault {
...
+            "VL: You are not authorized"
         );
         _;
     }
 
+    function authorize(address user) external onlyOwner {
+        allowed[user] = true;
+    }
+
+    function revokeAuth(address user) external onlyOwner {
```
Comment (Thomas-Smets) [inline src/Vault.sol:114]: Add tests

**Review by j-vp — COMMENTED** [review]
Add test that checks baseCurrency after vault deployment through factory, and general tests as in vault.t.sol through the factory as well (in end to end?)

**Comment by j-vp** [comment]
Can you add the refactor branch to https://github.com/arcadia-finance/lending-core/blob/main/.github/workflows/foundry.yml to trigger the tests? Maybe we should have it on all PR's anyway

File: src/Vault.sol [hunk 9]
```diff
@@ -50,12 +50,17 @@ contract Vault {
     address public _stakeContract;
     address public _irmAddress;
 
+
+
+    // ACCESS CONTROL
+    address public owner;
```
[the remark this answers is not in the document]
Comment (simluyt) [reply src/Vault.sol:56]: So this goes or stays?

[the remark this answers is not in the document]
Comment (j-vp) [reply src/Vault.sol:56]: Can stay!

File: src/Vault.sol [hunk 10]
```diff
@@ -658,10 +680,30 @@ contract Vault {
...
-            numeraire
+            baseCurrency
         );
     }
 
+    /** 
+    @notice Sets the baseCurrency of a vault.
+    @dev First checks if there is no locked value. If there is no value locked then the baseCurrency gets changed to the param
+  */
+    function setBaseCurrency(uint256 newBaseCurrency) public onlyAuthorized onlyOwner {
```
[the remark this answers is not in the document]
Comment (simluyt) [reply src/Vault.sol]: Good point

File: src/test/8_Vault.t.sol (hunk 7 above)
Comment (simluyt) [reply src/test/8_Vault.t.sol:1785]: I'll add in an extra test but this test is going to change within next two weeks when we change debt to valueLock ysstem?

File: src/Vault.sol (hunk 4 above)
Comment (simluyt) [reply src/Vault.sol:94]: This is about the modifier or the transfer Owner / liquidate methods?

Comment (simluyt) [reply src/Vault.sol:94]: Done except packed slot issue. Created a issue for this
